packages/backend/src/db/note/noteStore.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import * as Mongoose from 'mongoose';
+import { Model } from 'mongoose';
+import { Note, NoteDocument, NoteModel } from './note.schema';
+
+@Injectable()
+export class NoteStoreService {
+  constructor(
+    @InjectModel(Note.name)
+    private readonly noteModel: Model<NoteDocument>,
+  ) {
+    Mongoose.set('sanitizeFilter', true);
+  }
+
+  async create(createdNoteModel: NoteModel): Promise<NoteDocument> {
+    return this.noteModel.create(createdNoteModel);
+  }
+
+  async findOne(id: string | Mongoose.Types.ObjectId): Promise<NoteDocument> {
+    return this.noteModel.findOne({ _id: id }).exec();
+  }
+
+  async findAll(): Promise<NoteDocument[]> {
+    return this.noteModel.find().exec();
+  }
+
+  async findAllByHouse(
+    houseId: string | Mongoose.Types.ObjectId,
+  ): Promise<NoteDocument[]> {
+    return this.noteModel
+      .find()
+      .populate({ path: 'house', match: { _id: houseId } })
+      .exec();
+  }
+
+  async update(
+    id: string,
+    updateNoteModel: Partial<NoteModel>,
+  ): Promise<NoteDocument> {
+    return this.noteModel.findOneAndUpdate({ _id: id }, updateNoteModel).exec();
```

Comment:
This is currently returning the old document, consider doing something like this ` return this.noteModel.findOneAndUpdate({ _id: id }, updateNoteModel,
      {
        new: true,
      },).exec()` to return the updated document instead

Reply:
I say this because your update enpoint calls this function so it should probably return the updated document